src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,183 @@
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkElementIndex(index);
+        return getNodeOnIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkElementIndex(index);
+        Node<T> nodeOnIndex = getNodeOnIndex(index);
+        T changedValue = nodeOnIndex.item;
+        nodeOnIndex.item = value;
+        return changedValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkElementIndex(index);
+        return unLink(index);
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        int index = findIndexByValue(object);
+        if (index == -1) {
+            return false;
+        } else {
+            unLink(index);
+        }
+        return true;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void checkElementIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayIndexOutOfBoundsException("Index is not valid: " + index);
+        }
+    }
+
+    private int findIndexByValue(T value) {
+        Node<T> node = head;
+        for (int i = 0; i < size; i++) {
+            if (value == node.item || value != null && value.equals(node.item)) {
+                return i;
+            }
+            node = node.next;
+        }
+        return -1;
+    }
+
+    private Node<T> getNodeOnIndex(int index) {
+        Node<T> nodeOnIndex;
+        if (index < (size >> 1)) {
+            nodeOnIndex = head;
+            for (int i = 0; i < index; i++) {
+                nodeOnIndex = nodeOnIndex.next;
+            }
+        } else {
+            nodeOnIndex = tail;
+            for (int i = size - 1; i > index; i--) {
+                nodeOnIndex = nodeOnIndex.prev;
+            }
+        }
+        return nodeOnIndex;
+    }
+
+    private void addHeadNode(T value) {
+        Node<T> node = new Node<>(null, value, null);
+        head = node;
+        tail = node;
+    }
+
+    private void addNextNode(T value) {
+        Node<T> node = new Node<>(null, value, null);
+        if (head.equals(tail)) {
+            node.prev = head;
+            head.next = node;
+            head.prev = null;
+        } else {
+            node.prev = tail;
+            tail.next = node;
+        }
+        tail = node;
+    }
+
+    private void addNodeInside(int index, T value) {
+        Node<T> nextNode = getNodeOnIndex(index);
+        Node<T> node = new Node<>(nextNode.prev, value, nextNode);
+        nextNode.prev = node;
+        node.prev.next = node;
+        size++;
+    }
+
+    private T unLink(int index) {
```

Comment:
unlink, filename, database - it's just a single word, no need for uppercasing :)